static/js/recent_topics_ui.js

```diff
@@ -498,26 +574,43 @@ export function update_filters_view() {
     topics_widget.hard_redraw();
 }
 
-function stream_sort(a, b) {
-    const a_stream = message_store.get(a.last_msg_id).stream;
-    const b_stream = message_store.get(b.last_msg_id).stream;
-    if (a_stream > b_stream) {
+function sort_comparator(a, b) {
+    // compares strings in lowercase and returns -1, 0, 1
+    if (a.toLowerCase() > b.toLowerCase()) {
         return 1;
-    } else if (a_stream === b_stream) {
+    } else if (a === b) {
```

Comment:
This sorting algorithm is likely wrong; you can't do this comparison without `toLowerCase` when the above one was done with it.

`return util.strcmp(a.toLowerCase(), b.toLowerCase())` might be what you want to use instead of this function anyway.